src/npm/packagesFetcher.js

```diff
@@ -0,0 +1,148 @@
+import log from './../log.js';
+import wait from './../utils/wait.js';
+
+import * as api from './index.js';
+
+class PackagesFetcher {
+  limit = 100;
+  max = 1000;
+  total = 0;
+  storage = [];
```

Comment:
I guess it can but we are preloading max 1000 and we saw in heroku that we are on average at 200mb. The max can be tweaked to be more respectful of the limitation later ☺️